packages/editor-ui/src/components/ParameterInputFull.vue

```diff
@@ -53,7 +59,74 @@ const focused = ref(false);
 const menuExpanded = ref(false);
 const forceShowExpression = ref(false);
 
+const fromAiOverride = {
+	overridePlaceholder: 'Defined automatically by the model',
+	icon: 'plus',
+	output: '={{ [marker] $fromAI([key], [type], [default], [description]) }}',
+	providers: {
+		marker: '/* n8n-auto-generated-override */',
+		key: (props: Props) => props.parameter.name, // todo - need to check these
+		type: (props: Props) => 'string', // todo - infer boolean, number and other matching types
+	},
+	extraProps: {
+		description: {
+			default: '',
+			placeholder: 'Description of how the model should define this value',
+		},
+	},
+};
+
+function buildValueFromOverride(
+	providedExtraProps: (typeof fromAiOverride)['extraProps'], // todo generic-ify
+	includeMarker: boolean,
+) {
+	const {
+		providers: { marker, key, type },
+		extraProps,
+	} = fromAiOverride;
+	return `={{ ${includeMarker ? marker : ''} $fromAI('${key(props)}', '${providedExtraProps.description.default ?? extraProps.description}', '${type(props)}') }}`;
+}
+
+function isOverrideValue(s: string) {
+	return s.startsWith('={{ /* n8n-auto-generated-override */ $fromAI(');
+}
+
+function parseOverrides(s: string) {
+	/*
+		Approaches:
+		- Smart and boring: Reuse packages/core/src/CreateNodeAsTool.ts - unclear where to move it
```

Comment:
Maybe moving it to the workflow package, which a dependency of the the editor-ui package and then you can also use it here?

Reply:
editor-ui, core, and CLI use the workflow package. If only two of those packages use the functionality we can move it, as it's better than duplicating code.

Reply:
So far, it's not shared AFAIK. Unless I missing something.